### PR TITLE
RFR: Build out book information/quiz page

### LINKED ISSUE #62: FE: Build out book information/quiz page
**Is your feature request related to a problem? Please describe.**
We want a page that, given the route for the book id, will render the book information page, maybe a download link, and the quizzes to take. 

**Describe the solution you'd like**
screen shot 2018-10-29 at 9 27 21 pm

**Additional context**
Add details, any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by josh-byster**
## Status: 

:rocket: Ready

## Description
We want a page that, given the route for the book ID, will render the book information page, maybe a download link, and the quizzes to take.

Related issues: Closes #62 

### /book/:id
image

### /book/:id/:quizID
image

image

### CODE REVIEW

File: frontend/src/pages/BookPage.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      cards.push({
+        title: this.state.quizData[i]['name'],
+        id: this.state.quizData[i]['id'],
+        index: i
+      });
+    }
+    return cards;
+  };
+
+  selectQuiz = (id, index) => {
```
Comment (Davidw1339) [inline frontend/src/pages/BookPage.js]: not sure if easily possible but scrolling down to the newly opened quiz could be cool so the user isn't confused if they click the button and "nothing" happens

File: frontend/src/pages/BookPage.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  renderFunc = card => {
+    return (
+      <Button
+        color="success"
+        className="btn btn-block"
+        onClick={() => this.selectQuiz(card.id, card.index)}
+      >
+        {card.title}
+      </Button>
+      // <CatalogCard title={card.title} subtitle="" text="" onClickTitle="" />
```
Comment (Davidw1339) [inline frontend/src/pages/BookPage.js]: nit: remove comment :comet:

File: frontend/src/pages/BookPage.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      // TODO: remove in future
+      route: '/kids/catalog',
+      name: 'Catalog',
+      component: Catalog
+    }
+  ];
+
+  getCards = () => {
+    var cards = [];
+    for (var i in this.state.quizData) {
```
Comment (Davidw1339) [inline frontend/src/pages/BookPage.js]: nit: use let i

File: frontend/src/styles/BookInfo.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  color: #fff;
+  a {
+    color: #000;
+  }
+}
+
+.book-cover {
+  border-radius: 5px;
+  border: 2px solid white;
+  box-shadow: 5px 5px 5px 1px rgba(3, 15, 28, 0.331);
```
Comment (Davidw1339) [inline frontend/src/styles/BookInfo.scss]: oooo so fancy :black_circle: with the shadow

File: frontend/src/styles/BookInfo.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+}
+.blue-gradient {
```
Comment (Davidw1339) [inline frontend/src/styles/BookInfo.scss]: looking fresh :sparkle: :sparkler: :ocean:

File: frontend/src/utils/api.js [hunk 6]
```diff
@@ -50,5 +50,88 @@ function search(searchString) {
...
+function getBookData(id) {
+  // return fetch(`${API_URL}/books?id=${id}`, {
+  //   method: 'POST',
+  //   headers: {
+  //     'Content-Type': 'application/json'
+  //   },
+  //   body: JSON.stringify(id)
+  // }).then(res => res.json());
+
+  const mockAPI = {
```
Comment (Davidw1339) [inline frontend/src/utils/api.js]: hopefully can replace with backend api to test integration
you may need to traverse one layer deeper in the results of the api call :calling: 
lemme know if backend stuff is being weird

File: frontend/src/components/QuizViewer.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+import Question from '../components/Question';
+import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+class QuizViewer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (Davidw1339) [inline frontend/src/components/QuizViewer.js]: eventually may have to save :floppy_disk:  the state of previously completed questions for consistency

File: frontend/src/components/QuizViewer.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      currentQuestion: 0
+    };
+  }
+  getQuestionObject() {
+    return this.props.questionList[this.state.currentQuestion];
```
[the remark this answers is not in the document]
Comment (XuanLi-Leong) [reply frontend/src/components/QuizViewer.js]: I wonder if we should have each quiz on a new page, for ease of returning to the quiz

File: frontend/src/styles/BookPage.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
+.catalog {
```
[the remark this answers is not in the document]
Comment (josh-byster) [reply frontend/src/styles/BookPage.scss]: Good catch! Thanks!

File: frontend/src/components/Catalog.js [hunk 10]
```diff
@@ -101,7 +101,15 @@ class Catalog extends Component {
   render() {
     return (
       <div>
-        <Container fluid={true} className="catalog">
+        <Container
+          fluid={true}
+          className={
+            'catalog' +
+            (this.props.className !== undefined
```
Comment (XuanLi-Leong) [inline frontend/src/components/Catalog.js]: Awesome stuff

File: frontend/src/components/Question.js [hunk 11]
```diff
@@ -74,25 +66,16 @@ class Question extends Component {
     let input = (
       <Input
         key={i}
-        onClick={() => this.setState({ selectedAnswer: i })}
+        onClick={() => this.props.setQuestionProps({ selectedAnswer: i })}
```
Comment (Davidw1339) [inline frontend/src/components/Question.js]: state elevation looked like a pain :face_with_head_bandage: good work tho!

File: frontend/src/components/QuizViewer.js [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+import Question from '../components/Question';
+import { Button, Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+class QuizViewer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      questionProps: this.props.questionList.map(question => {
+        return {
+          selectedAnswer: -1,
```
Comment (Davidw1339) [inline frontend/src/components/QuizViewer.js]: good work with encapsulating all question state in here

File: frontend/src/pages/BookPage.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+    this.fetchBookData();
+    this.fetchQuizData();
+  }
+
+  fetchBookData = async () => {
+    const { message, success, result } = await getBookData(
+      this.props.match.params.id
+    );
+    if (success) {
+      this.setState({ bookData: result['results'][0] });
```
Comment (Davidw1339) [inline frontend/src/pages/BookPage.js]: do we need some check to make sure the book data exists ex: result['results'] has length >= 1?

File: frontend/src/pages/BookPage.js [hunk 14]
```diff
@@ -22,7 +22,13 @@ class BookPage extends Component {
       this.props.match.params.id
     );
     if (success) {
-      this.setState({ bookData: result['results'][0] });
+      if (result['results'].length > 0) {
+        this.setState({ bookData: result['results'][0] });
```
Comment (Davidw1339) [inline frontend/src/pages/BookPage.js]: dang great stuff :rocket:

**Pull request merged**